sdk/search/search/review/search.api.md

```diff
@@ -246,11 +817,113 @@ export type SuggestResult<T> = {
+    expand?: boolean;
+    ignoreCase?: boolean;
+    name: string;
+    odatatype: "#Microsoft.Azure.Search.SynonymTokenFilter";
+    synonyms: string[];
+}
+
+// @public
+export interface TagScoringFunction {
+    boost: number;
+    fieldName: string;
+    interpolation?: ScoringFunctionInterpolation;
+    parameters: TagScoringParameters;
+    type: "tag";
+}
+
+// @public
+export interface TagScoringParameters {
+    tagsParameter: string;
+}
+
+// @public
+export interface TextWeights {
+    weights: {
+        [propertyName: string]: number;
+    };
+}
+
+// @public
+export type TokenCharacterKind = 'letter' | 'digit' | 'whitespace' | 'punctuation' | 'symbol';
+
+// @public
+export interface TokenFilter {
+    name: string;
+    odatatype: "TokenFilter";
+}
+
+// @public
+export type TokenFilterName = 'arabic_normalization' | 'apostrophe' | 'asciifolding' | 'cjk_bigram' | 'cjk_width' | 'classic' | 'common_grams' | 'edgeNGram_v2' | 'elision' | 'german_normalization' | 'hindi_normalization' | 'indic_normalization' | 'keyword_repeat' | 'kstem' | 'length' | 'limit' | 'lowercase' | 'nGram_v2' | 'persian_normalization' | 'phonetic' | 'porter_stem' | 'reverse' | 'scandinavian_normalization' | 'scandinavian_folding' | 'shingle' | 'snowball' | 'sorani_normalization' | 'stemmer' | 'stopwords' | 'trim' | 'truncate' | 'unique' | 'uppercase' | 'word_delimiter';
+
+// @public
+export type TokenFilterUnion = TokenFilter | AsciiFoldingTokenFilter | CjkBigramTokenFilter | CommonGramTokenFilter | DictionaryDecompounderTokenFilter | EdgeNGramTokenFilter | EdgeNGramTokenFilterV2 | ElisionTokenFilter | KeepTokenFilter | KeywordMarkerTokenFilter | LengthTokenFilter | LimitTokenFilter | NGramTokenFilter | NGramTokenFilterV2 | PatternCaptureTokenFilter | PatternReplaceTokenFilter | PhoneticTokenFilter | ShingleTokenFilter | SnowballTokenFilter | StemmerTokenFilter | StemmerOverrideTokenFilter | StopwordsTokenFilter | SynonymTokenFilter | TruncateTokenFilter | UniqueTokenFilter | WordDelimiterTokenFilter;
+
+// @public
+export interface TokenInfo {
+    readonly endOffset: number;
+    readonly position: number;
+    readonly startOffset: number;
+    readonly token: string;
+}
+
+// @public
+export interface Tokenizer {
+    name: string;
+    odatatype: "Tokenizer";
+}
+
+// @public
+export type TokenizerName = 'classic' | 'edgeNGram' | 'keyword_v2' | 'letter' | 'lowercase' | 'microsoft_language_tokenizer' | 'microsoft_language_stemming_tokenizer' | 'nGram' | 'path_hierarchy_v2' | 'pattern' | 'standard_v2' | 'uax_url_email' | 'whitespace';
+
+// @public
+export type TokenizerUnion = Tokenizer | ClassicTokenizer | EdgeNGramTokenizer | KeywordTokenizer | KeywordTokenizerV2 | MicrosoftLanguageTokenizer | MicrosoftLanguageStemmingTokenizer | NGramTokenizer | PathHierarchyTokenizerV2 | PatternTokenizer | StandardTokenizer | StandardTokenizerV2 | UaxUrlEmailTokenizer;
+
+// @public
+export interface TruncateTokenFilter {
+    length?: number;
```

Comment:
It seems like this defaults to 300. Is that what people usually want (e.g. because they're trimming tokens produced by a custom analyzer)? Lucene docs suggest it also gets with much smaller values for indexing turkish, though I don't understand why. Anyway, if people will in practice almost always be customizing this value, making it required would save us from having to document the default and save code readers from having to look it up when the value is omitted.

Reply:
Good question. @Yahnoosh hopefully has a more informed perspective on this than I do.

Reply:
Yes, you're right. We don't allow tokens that are longer than 300 characters to be produced by custom analyzers, that's why the default is equal to the max value. Lucene docs mention that the truncating 5 first characters is a successful stemming strategy for Turkish but that's unrelated (I think that comment is misplaced). I ran a query that shows 99% of those token filters in production used the default. 

@brjohnstmsft, for your reference:

IndexMetadataSnapshot
| where TIMESTAMP > ago(1d) 
| where IndexMetadataComponent == "Definition.TokenFilters"
| mv-expand IndexMetadata
| where IndexMetadata.type == "truncate"
| summarize count() by toint(IndexMetadata.options.length)